File: src/main/java/com/bazbatlabs/smashballs/models/Wall.java
package com.bazbatlabs.smashballs.models;

public final class Wall implements Collidable {

    private final Rect bounds;
    private final WorldEvents events;

    public Wall(Rect bounds, WorldEvents events) {
        this.bounds = bounds;
        this.events = events;
    }

    public Rect bounds() { return bounds; }

    public void hit() {
        events.enqueue(WorldEvents.Event.WALL_HIT);
    }

    public Vec2 deflect(Vec2 vel, Vec2 collision, Axis axis) {
        switch (axis) {
        case X: return new Vec2(-vel.x, vel.y);
        case Y: return new Vec2(vel.x, -vel.y);
        default: return Vec2.ZERO;
        }
    }

    public boolean isActive() { return true; }
}

File: src/main/scala/models/Wall.scala
package com.bazbatlabs.bricksmash.models

class Wall(val bounds: Rect, private val events: WorldEvents) extends Collidable {

  def hit() {
    events.enqueue(Event.WallHit)
  }

  def deflect(vel: Vec2, collision: Vec2, axis: Axis.Value) = {
    axis match {
      case Axis.X => new Vec2(-vel.x, vel.y)
      case Axis.Y => new Vec2(vel.x, -vel.y)
      case _ => Vec2.Zero
    }
  }

  def isActive = true
}
